pkg/detectors/postgres/postgres.go

```diff
@@ -0,0 +1,264 @@
+	var results []detectors.Result
+	var pgURLs []url.URL
+	pgURLs = append(pgURLs, findUriMatches(string(data)))
+	pgURLs = append(pgURLs, findComponentMatches(verify, string(data))...)
+
+	for _, pgURL := range pgURLs {
+		if pgURL.User == nil {
+			continue
+		}
+		username := pgURL.User.Username()
+		password, _ := pgURL.User.Password()
+		hostport := pgURL.Host
+		result := detectors.Result{
+			DetectorType: detectorspb.DetectorType_Postgres,
+			Raw:          []byte(username + password),
+			RawV2:        []byte(hostport + username + password),
+		}
+
+		if verify {
+			timeoutInSeconds := getDeadlineInSeconds(ctx)
+			isVerified, verificationErr := verifyPostgres(&pgURL, timeoutInSeconds)
+			result.Verified = isVerified
+			result.SetVerificationError(verificationErr, password)
+		}
+
+		if !result.Verified && detectors.IsKnownFalsePositive(password, detectors.DefaultFalsePositives, true) {
+			continue
+		}
+		results = append(results, result)
+	}
+
+	return results, nil
+}
+
+func getDeadlineInSeconds(ctx context.Context) int {
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		// Context does not have a deadline
+		return 0
+	}
+
+	duration := time.Until(deadline)
+	return int(duration.Seconds())
+}
+
+func findUriMatches(dataStr string) url.URL {
+	var pgURL url.URL
+	for _, uri := range uriPattern.FindAllString(dataStr, -1) {
+		pgURL, err := url.Parse(uri)
+		if err != nil {
+			continue
+		}
+		if pgURL.User != nil {
+			return *pgURL
+		}
+	}
+	return pgURL
+}
+
+// check if postgres is running
+func postgresRunning(hostname, port string) bool {
+	connStr := fmt.Sprintf("host=%s port=%s sslmode=disable", hostname, port)
+	db, err := sql.Open("postgres", connStr)
+	if err != nil {
+		return false
+	}
+	defer db.Close()
+	return true
+}
+
+func findComponentMatches(verify bool, dataStr string) []url.URL {
+	usernameMatches := common.UsernameRegexCheck("").Matches([]byte(dataStr))
+	passwordMatches := common.PasswordRegexCheck("").Matches([]byte(dataStr))
+	hostnameMatches := hostnamePattern.FindAllStringSubmatch(dataStr, -1)
+	portMatches := portPattern.FindAllStringSubmatch(dataStr, -1)
+
+	var pgURLs []url.URL
+
+	hosts := findHosts(verify, hostnameMatches, portMatches)
+
+	for _, username := range dedupMatches(usernameMatches) {
+		for _, password := range dedupMatches(passwordMatches) {
+			for _, host := range hosts {
+				hostname, port := strings.Split(host, ":")[0], strings.Split(host, ":")[1]
+				if combinedLength := len(username) + len(password) + len(hostname); combinedLength > 255 {
+					continue
+				}
+				postgresURL := url.URL{
+					Scheme: "postgresql",
+					User:   url.UserPassword(username, password),
+					Host:   fmt.Sprintf("%s:%s", hostname, port),
+				}
+				pgURLs = append(pgURLs, postgresURL)
+			}
+		}
+	}
+	return pgURLs
+}
+
+// if verification is turned on, and we are able to verify at least one host, return only verified hosts
```

Comment:
you're overloading "verify" here in that you're using it to mean "postgres is running at all" rather than "we found valid credentials". i'd find it easier to read if you reworded the comment to not use the word "verify", e.g. `if verification is turned on, and we can confirm that postgres is running on at least one host, return only hosts where it's running. otherwise return all hosts.`